tests/cephadm/test_cephadm_custom_config_support.py

```diff
@@ -0,0 +1,55 @@
+from cli.cephadm.cephadm import CephAdm
+from cli.exceptions import CephadmOpsExecutionError
+from cli.ops.cephadm import generate_bootstrap_config
+from cli.utilities.containers import Container
+
+
+def check_custom_file(node, config):
+    """Check custom file updated to conatiner"""
+    for spec in config["spec"]:
+        service = spec["service_name"]
+        container_id = (
+            Container(node)
+            .ps(filter={"name": service})[0]
+            .split("\n")[1]
+            .split(" ")[0]
+            .strip()
+        )
+        mount_path = spec["custom_configs"][1]["mount_path"]
+        result = node.exec_command(
+            sudo=True, cmd=f"podman exec -ti {container_id} cat {mount_path}"
+        )[0]
+        if result != spec["custom_configs"][1]["content"]:
+            raise CephadmOpsExecutionError(
+                "{service} container file content not updated"
+            )
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify cephadm custom config file support"""
+    # Get config specs
+    config = kwargs.get("config")
+
+    # Get cluster installer node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Generate custom config file
+    config_file = generate_bootstrap_config(installer, config)
+
+    # Mount custom config file
+    conf = {"pos_args": []}
+    out = CephAdm(installer, config_file).ceph.orch.apply(input=config_file, **conf)
+    if "Scheduled" not in out:
+        raise CephadmOpsExecutionError("Fail to mount custom config file")
+
+    # Redeploy services
+    for spec in config["spec"]:
```

Comment:
I would prefer `dict.get` instead of using square brackets `[]`

Reply:
Thanks @vamahaja 
Updated